.github/workflows/test-gpu.yml

```diff
@@ -0,0 +1,43 @@
+name: Test GPU
+
+on: [pull_request]
+
+jobs:
+  build-test:
+    runs-on: [self-hosted, gpu]
+    strategy:
+      matrix:
+        include:
+          - itk-git-tag: "c693dbc830859ea10d573301f075e121a4c6c581"
+            cmake-build-type: "MinSizeRel"
+        
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Download ITK
+      run: |
+        pushd ~/ITK
+        git checkout ${{ matrix.itk-git-tag }}
```

Comment:
git clone ITK missing.

Reply:
We should be using the `actions/checkout` action to checkout the ITK repository both here and in the ITKModuleTemplate.

Reply:
To clarify what's going on here, we're relying on a custom Ubuntu 18.04 image we create from an EC2 node with the boilerplate code we need and going from there. For cleanliness I'll remove the ITK folder already cloned on the instance and clone at CI build time instead.

Boilerplate that we definitely want pre-installed:
- OpenCL drivers (in this case for Ubuntu + AMD GPU)

Boilerplate that is pre-installed and can stay that way to minimize node runtime:
- CMake, GNinja, gcc
- OpenCL-ICD-Loader
- OpenCL-Headers

Source trees to be cloned and built at runtime:
- ITK
- ITKVkFFTBackend (w/ testing)